iidm/iidm-serde/src/test/java/com/powsybl/iidm/serde/extensions/ActivePowerControlXmlTest.java

```diff
@@ -44,13 +46,36 @@ public void setUp() throws IOException {
     }
 
     @Test
-    void test() throws IOException {
-        Network network2 = allFormatsRoundTripTest(network, "/activePowerControlRoundTripRef.xml", CURRENT_IIDM_VERSION);
+    void testPLimitOverride() throws IOException {
+        network.getGenerator("GEN").getExtension(ActivePowerControl.class).setMaxPOverride(100.);
+        network.getBattery("BAT").getExtension(ActivePowerControl.class).setMinPOverride(10.);
+        Network network2 = allFormatsRoundTripTest(network, "/activePowerControlWithLimitRoundTripRef.xml", CURRENT_IIDM_VERSION);
+
+        Generator gen2 = network2.getGenerator("GEN");
+        assertNotNull(gen2);
+        ActivePowerControl<Generator> activePowerControl1 = gen2.getExtension(ActivePowerControl.class);
+        assertNotNull(activePowerControl1);
+        assertEquals(100, activePowerControl1.getMaxPOverride().get());
+        assertTrue(activePowerControl1.getMinPOverride().isEmpty());
+
+        Battery bat2 = network2.getBattery("BAT");
+        assertNotNull(bat2);
+        ActivePowerControl<Battery> activePowerControl2 = bat2.getExtension(ActivePowerControl.class);
+        assertNotNull(activePowerControl2);
+        assertTrue(activePowerControl2.getMaxPOverride().isEmpty());
+        assertEquals(10, activePowerControl2.getMinPOverride().get());
```

Comment:
To avoid the `get()` which might be seen as a code smell you could do so:
```suggestion
        assertEquals(OptionalDouble.of(10), activePowerControl2.getMinPOverride());
```

Reply:
Done. But don't see a strong smell difference, especially in an assert. Makes the expected more complex to me.

Reply:
if empty the result of the test is a bit clearer that way